src/HearThis/UI/SentenceClauseSplitter.cs

```diff
@@ -51,6 +51,11 @@ public SentenceClauseSplitter(char[] additionalSeparators)
 			_breakAtFirstLevelQuotes = breakAtFirstLevelQuotes && _firstLevelStartQuotationMark != null;
 			if (_breakAtFirstLevelQuotes)
 				Debug.Assert(_firstLevelEndQuotationMark != null);
+			if (string.IsNullOrEmpty(_firstLevelStartQuotationMark) || string.IsNullOrEmpty(_firstLevelEndQuotationMark))
```

Comment:
I suppose we should test that but the infinite loop was caused by adding the length of the string to a variable and expecting that to advance the loop.